public/pages/Main/Main.tsx

```diff
@@ -32,62 +44,155 @@ enum Pathname {
+        availableConfigTypes: [],
+        tooltipSupport: false,
+      };
+    }
   }
 
   async componentDidMount() {
-    const serverFeatures = await this.context.notificationService.getServerFeatures();
-    if (serverFeatures != null) {
-      this.setState({
-        availableChannels: serverFeatures.availableChannels,
-        availableConfigTypes: serverFeatures.availableConfigTypes,
-        tooltipSupport: serverFeatures.tooltipSupport,
-      });
-    } else {
-      // Feature API call failed, allow all configs to avoid UI breaking.
-      // User requests will still be validated by backend.
+
+    const services = this.getServices(this.props.http) // Assuming this.context holds the value provided by ServicesContext
+    const serverFeatures = await services.notificationService.getServerFeatures();
+
+    if (this.props.multiDataSourceEnabled) {
+      if (serverFeatures != null) {
+        this.setState({
+          dataSourceId: this.state.dataSourceId, dataSourceLabel: this.state.dataSourceLabel, dataSourceReadOnly: false,
+          availableChannels: serverFeatures.availableChannels,
+          availableConfigTypes: serverFeatures.availableConfigTypes,
+          tooltipSupport: serverFeatures.tooltipSupport
+        });
+      } else {
+        // Feature API call failed, allow all configs to avoid UI breaking.
+        // User requests will still be validated by backend.
+        this.setState({
+          dataSourceId: this.state.dataSourceId, dataSourceLabel: this.state.dataSourceLabel, dataSourceReadOnly: false,
+          availableChannels: CHANNEL_TYPE,
+          availableConfigTypes: [
+            'slack',
+            'chime',
+            'microsoft_teams',
+            'webhook',
+            'email',
+            'sns',
+            'smtp_account',
+            'ses_account',
+            'email_group',
+          ],
+          tooltipSupport: false
+        });
+      }
+    }
+    else
+      {
+        if (serverFeatures != null) {
+          this.setState({
+            availableChannels: serverFeatures.availableChannels,
+            availableConfigTypes: serverFeatures.availableConfigTypes,
+            tooltipSupport: serverFeatures.tooltipSupport
+          });
+        } else {
+          // Feature API call failed, allow all configs to avoid UI breaking.
+          // User requests will still be validated by backend.
+          this.setState({
+            availableChannels: CHANNEL_TYPE,
+            availableConfigTypes: [
+              'slack',
+              'chime',
+              'microsoft_teams',
+              'webhook',
+              'email',
+              'sns',
+              'smtp_account',
+              'ses_account',
+              'email_group',
+            ],
+            tooltipSupport: false
+          });
+        }
+      }
+  }
+
+  onSelectedDataSources = (dataSources: DataSourceOption[]) => {
+    const { id = "", label = "" } = dataSources[0] || {};
```

Comment:
Do we need a null check at the `dataSources[0]`, or do we never expect `dataSources` to be null, or an empty array?